documentation/add_compiler.md

```diff
@@ -0,0 +1,33 @@
+# Add Compiler to support PCP
+
+The peadm::add_compler plan can be used to add a new compiler to a PE architecture or replace an existing one with new configuration.
+
+## Add compiler to an exising PE instance
+
+Prepare to run the plan using a params.json file such as this one:
+
+```json
+{
+  "avail_group_letter": "A",
+  "compiler_host": "pe-xl-compiler-0.lab1.puppet.vm",
+  "dns_alt_names": [ "puppet", "puppet.lab1.puppet.vm" ],
+  "primary_host":  "pe-xl-core-0.lab1.puppet.vm",
+  "primary_postgresql_host": "pe-psql-6251cd-0.us-west1-a.c.slice-cody.internal",
+}
+```
+
+See the [install](install.md#reference-architectures) documentation for a list of supported architectures.
+
+
+## Running the add_compiler plan
+```
+bolt plan run peadm::add_compiler --params @params.json 
+```
+
+This call will retreive the current peadm config to determain the setup rules needed for a compiler's secondary PuppetDB instances. The provided server will be configured with the appropriate rules for Puppet Server access from compiler. The puppet.service will be stopped and the pe-postgresql.service will be reloaded. If required and agent ewill be installed and regenerated agent certificate to add required data with peadm::subplans::component_install. Puppet agent will run on the following components;
+* _\<compiler-host\>_
+* _\<primary_postgresql_host\>_
+* _\<replica puppetdb\>_
```

Comment:
Is that the replica host?

Reply:
from the add_compiler.pp
`  # On replica puppetdb run the puppet agent
  run_task('peadm::puppet_runonce', $replica_puppetdb_target)`

Reply:
yes, so that an incorrect wording. There is no such thing as "replica puppetdb". 
We do have "replica postgres host" or just the "primary replica host" - I'm assuming the former is meant here.

Reply:
Thank you